### DISCUSSION TITLE
Different OIDC configuration for single vhosts

### DISCUSSION THREAD

**Discussion by Sugavanam** [OpenIDC/mod_auth_openidc#1008] [General]
Hello 
I have a scenario where I require, different OIDC configs based on the request at runtime, Is there any solution to handle the configs conditional based at runtime

<VirtualHost *:80>
        OIDCSessionType client-cookie
        OIDCClientID myclient-id
        OIDCClientSecret XXXXXXXXXXXXXXXXXXXXXXXXXX
        OIDCProviderMetadataURL https://auth.example.com/.well-known/openid-configuration
        OIDCProviderAuthorizationEndpoint https://auth.example.com/oAuth2/authorize?values=L4
        OIDCRedirectURI /redirect_uri
        OIDCCryptoPassphrase xxxxxxxxxxxx
 </VirtualHost>
 
For example if we have different OIDCProviderAuthorizationEndpoint https://auth.example.com/oAuth2/authorize?values=L2 
Is there a way to change the loaded configs based on the request?

**Comment by zandbelt** [comment]
see https://github.com/zmartzone/mod_auth_openidc/wiki#13-how-can-i-add-custom-parameters-to-the-authorization-request, last paragraph

**Reply by Sugavanam** [reply]
Thanks it works for me.
And also the previous [discussion ](https://github.com/zmartzone/mod_auth_openidc/issues/281)for path separated application helped a lot
